modules/apps/portal-language/portal-language-override-service/src/main/java/com/liferay/portal/language/override/internal/PLOEntryModelListener.java

```diff
@@ -30,23 +31,38 @@ public class PLOEntryModelListener extends BaseModelListener<PLOEntry> {
 
 	@Override
 	public void onAfterCreate(PLOEntry ploEntry) {
-		_updatePLOLanguageOverrideProvider(MethodType.ADD, ploEntry);
+		TransactionCommitCallbackUtil.registerCallback(
+			() -> {
+				_updatePLOLanguageOverrideProvider(MethodType.ADD, ploEntry);
 
-		_notifyCluster(MethodType.ADD, ploEntry);
+				_notifyCluster(MethodType.ADD, ploEntry);
+
+				return null;
+			});
 	}
 
 	@Override
 	public void onAfterRemove(PLOEntry ploEntry) {
-		_updatePLOLanguageOverrideProvider(MethodType.REMOVE, ploEntry);
+		TransactionCommitCallbackUtil.registerCallback(
+			() -> {
+				_updatePLOLanguageOverrideProvider(MethodType.REMOVE, ploEntry);
+
+				_notifyCluster(MethodType.REMOVE, ploEntry);
 
-		_notifyCluster(MethodType.REMOVE, ploEntry);
+				return null;
+			});
 	}
 
 	@Override
 	public void onAfterUpdate(PLOEntry originalPLOEntry, PLOEntry ploEntry) {
-		_updatePLOLanguageOverrideProvider(MethodType.UPDATE, ploEntry);
+		TransactionCommitCallbackUtil.registerCallback(
+			() -> {
+				_updatePLOLanguageOverrideProvider(MethodType.UPDATE, ploEntry);
+
+				_notifyCluster(MethodType.UPDATE, ploEntry);
 
-		_notifyCluster(MethodType.UPDATE, ploEntry);
+				return null;
+			});
```

Comment:
If possible, we should have test cases for all of the changes where we added the transaction callback support, so `onAfterCreate` (done), `onAfterRemove`, and `onAfterUpdate`.

Look for usages of `TransactionInvokerUtil.invoke` in test classes to see how to create your own transaction boundaries. That way, you can invoke a method, throw an exception, and then assert the expected results.

Reply:
I added tests for each event case. But I don't understand the necessity of using `TransactionInvokerUtil.invoke` for this scenario

Reply:
The idea is to simulate an exception being thrown from inside a transaction boundary. Do the tests fail if you revert the transaction changes to the ModelListener?

Reply:
Yes, if I remove the changes made to the ModelListener the test fails